lib/server.js

```diff
@@ -414,16 +414,17 @@ class Server extends EventEmitter {
   /*
    * Retrieve the ismaster for this server
    */
-  ismaster() {
+  ismaster(options) {
```

Comment:
I see you're adding `options` to `ismaster` here, but it looks like these should perhaps be stored in the `self.options`, or even the `clientOptions`?

Reply:
The reason why I added them here is because these two parameters are a connection by connection specification. I could see a person wanting to increase the connection or socket timeout on just one command because it runs longer or something. 

If I add these options to `self.options` or `clientOptions`, then do I use these parameters in the other spots as well, such as `executeCommand`?

Reply:
Between `self.options` and `self.clientOptions`, the better place would be `self.clientOptions` in my opinion because this is a timeout for the client itself.

If we put these parameters in `self.clientOptions`, then it all works fine but if the user wants to set a specific timeout for just one command, they have to do something like:
```
server.clientOptions.socketTimeoutMS = 15000;
yield server.executeCommand(...);
delete server.clientOptions.socketTimeoutMS; // Or however you delete from an object in JS
```

Reply:
Either way works for me, I'll let you make the decision. It just depends how you think people will use this option most.

Reply:
We run into this a lot with the driver, so I think it bears consideration here. If we end up supporting options on each command, then we must be very careful to always "resolve" the option accordingly. I'm fine adding options here, as long as we take an approach to generating the "client configuration" I mentioned above: a common method that takes into account the client options provided in the constructor as the default values in the absence of one provided at the command level